core/loader/json-feed.ts

```diff
@@ -91,19 +91,34 @@ function validate<ValidatedType>(
   return true
 }
 
+const IMG_URL_REGEXP = /<img[^>]+src="?([^"\s]+)"?\s*\/?>/g
+
 function parsePosts(text: TextResponse): OriginPost[] {
   let parsedJson = text.parseJson()
   if (!validate<JsonFeed>(parsedJson, JSON_FEED_VALIDATORS)) return []
 
-  return parsedJson.items.map(item => ({
-    full: (item.content_html || item.content_text) ?? undefined,
-    intro: item.summary ?? undefined,
-    media: [],
-    originId: item.id,
-    publishedAt: toTime(item.date_published) ?? undefined,
-    title: item.title,
-    url: item.url ?? undefined
-  }))
+  return parsedJson.items.map(item => {
+    const full = (item.content_html || item.content_text) ?? undefined
+    const allImages: string[] = [item.banner_image ?? '', item.image ?? '']
+
+    if (full) {
+      let match: RegExpExecArray | null
+      while ((match = IMG_URL_REGEXP.exec(full))) {
+        const url = match[1]
+        url && allImages.push(url)
+      }
+    }
+
+    return {
+      full,
+      intro: item.summary ?? undefined,
+      media: [...new Set(allImages.filter(url => !!url))],
```

Comment:
Do we need to remove duplicates?

Reply:
https://www.jsonfeed.org/version/1/ says
```text
image (optional, string) is the URL of the main image for the item. This image may also appear in the content_html — if so, it’s a hint to the feed reader that this is the main, featured image. Feed readers may use the image as a preview (probably resized as a thumbnail and placed in a timeline).
```

So I suspect there may be duplicates

Reply:
Can you create a `unique()` function in `loader/utils.ts`?

Not everyone is knowing the `new Set()` hack. Helpers can improve readability by black-boxing hacks.

Also `unique()` can clean `''` strings.

Reply:
Or I can help if it is too boring :)

Reply:
Updated